### PR TITLE
All build functions use nas_security.Encode()

### PR DESCRIPTION
**Pull request by ShouheiNishi**
Some NAS messages are always sent as plain messages regardless of security context is valid.
This is incorrect behavior that violates NAS specifications.
This PR fix this.

### CODE REVIEW

**Comment by brianchennn** [comment]
@ShouheiNishi
Does this PR completely fix the [free5gc issue #196](https://github.com/free5gc/free5gc/issues/196) ?

**Comment by ShouheiNishi** [comment]
> @ShouheiNishi Does this PR completely fix the [free5gc issue #196](https://github.com/free5gc/free5gc/issues/196) ?

No this PR does not fix this issue.

File: internal/gmm/message/build.go [hunk 1]
```diff
@@ -165,7 +165,11 @@ func BuildAuthenticationRequest(ue *context.AmfUe) ([]byte, error) {
 
 	m.GmmMessage.AuthenticationRequest = authenticationRequest
 
-	return m.PlainNasEncode()
+	m.SecurityHeader = nas.SecurityHeader{
+		ProtocolDiscriminator: nasMessage.Epd5GSMobilityManagementMessage,
+		SecurityHeaderType:    nas.SecurityHeaderTypeIntegrityProtectedAndCiphered,
+	}
+	return nas_security.Encode(ue, m, accessType)
```
Comment (ss920386) [inline internal/gmm/message/build.go:172]: Will this make UE get a plain-nas message while nas sec header is SecurityHeaderTypeIntegrityProtectedAndCiphered when secCtx is not available?

Comment (ShouheiNishi) [reply internal/gmm/message/build.go:172]: No, when security context is not exist, `nas_security.Encode()` use SpareHalfOctetAndSecurityHeaderType as security header type  and `m.SecurityHeader` is not used.

Comment (ss920386) [reply internal/gmm/message/build.go:172]: Get it, thanks.

**Pull request merged**